var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -122,72 +142,106 @@ def check_variants(self, spec):
+                'METIS_OPT=-I%s' % spec['parmetis'].prefix.include,
+                'METIS_LIB=%s' % metis_lib]
             superlu_lib = '-L%s' % spec['superlu-dist'].prefix.lib
             superlu_lib += ' -lsuperlu_dist'
-            sl_inc = 'SUPERLU_OPT=-I%s' % spec['superlu-dist'].prefix.include
-            options.extend(['MFEM_USE_SUPERLU=YES',
-                            'SUPERLU_DIR=%s' % spec['superlu-dist'].prefix,
-                            sl_inc,
-                            'SUPERLU_LIB=%s' % superlu_lib])
+            options += [
+                'SUPERLU_DIR=%s' % spec['superlu-dist'].prefix,
+                'SUPERLU_OPT=-I%s' % spec['superlu-dist'].prefix.include,
+                'SUPERLU_LIB=%s' % superlu_lib]
+        else:
+            metis_lib = '-L%s -lmetis' % spec['metis'].prefix.lib
+            options += [
+                'METIS_DIR=%s' % spec['metis'].prefix,
+                'METIS_OPT=-I%s' % spec['metis'].prefix.include,
+                'METIS_LIB=%s' % metis_lib]
 
         if '+suite-sparse' in spec:
             ssp = spec['suite-sparse'].prefix
             ss_lib = '-L%s' % ssp.lib
-
             if '@3.2:' in spec:
                 ss_lib += ' -lklu -lbtf'
-
             ss_lib += (' -lumfpack -lcholmod -lcolamd' +
                        ' -lamd -lcamd -lccolamd -lsuitesparseconfig')
-
             no_rt = spec.satisfies('platform=darwin')
             if not no_rt:
                 ss_lib += ' -lrt'
             ss_lib += (' ' + metis_lib + ' ' + lapack_lib)
-
-            options.extend(['MFEM_USE_SUITESPARSE=YES',
-                            'SUITESPARSE_DIR=%s' % ssp,
-                            'SUITESPARSE_OPT=-I%s' % ssp.include,
-                            'SUITESPARSE_LIB=%s' % ss_lib])
+            options += [
+                'SUITESPARSE_DIR=%s' % ssp,
+                'SUITESPARSE_OPT=-I%s' % ssp.include,
+                'SUITESPARSE_LIB=%s' % ss_lib]
+
+        if '+sundials' in spec:
+            sundials_libs = (
+                '-lsundials_arkode -lsundials_cvode'
+                ' -lsundials_nvecserial -lsundials_kinsol')
+            if '+mpi' in spec:
+                sundials_libs += (
+                    ' -lsundials_nvecparhyp -lsundials_nvecparallel')
+            options += [
+                'SUNDIALS_DIR=%s' % spec['sundials'].prefix,
+                'SUNDIALS_OPT=-I%s' % spec['sundials'].prefix.include,
+                'SUNDIALS_LIB=-L%s %s' % (spec['sundials'].prefix.lib,
+                                          sundials_libs)]
+
+        if '+petsc' in spec:
+            f = open('%s/lib/pkgconfig/PETSc.pc' % spec['petsc'].prefix, 'r')
+            for line in f:
+                if re.search('^\s*Cflags: ', line):
+                    petsc_opts = re.sub('^\s*Cflags: (.*)', '\\1', line)
+                elif re.search('^\s*Libs.*: ', line):
+                    petsc_libs = re.sub('^\s*Libs.*: (.*)', '\\1', line)
+            f.close()
```

Comment:
There's a much easier way to do this. Check out the `stream` package for an example of `FileFilter`. Also see [the documentation](http://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.filter_file) for valid options.